README.md

```diff
@@ -122,4 +122,20 @@ The `KeyVaultCA.Web` writes logs to an Azure Application Insights instance, for
 
 The Terraform scripts listed under the `terraform` directory can be used to deploy the infrastructure required for E2E testing to an Azure environment. This deployment includes an App Service for the EST server to run in using an image pulled from Azure Container Registry, an Azure Key Vault for storing the Root CA and an IoT Hub, Device Provisioning Service and a Linux VM simulating an IoT Edge device. The Terraform template uses `dotnet run` to execute the API Facade Console App, hence installing the [.NET Runtime 6](https://dotnet.microsoft.com/en-us/download/dotnet/6.0) is required. The infrastructure can be deployed by cd'ing into the `terraform` directory and then running `terraform init` followed by `terraform apply`. Terraform will use the [logged in Azure user credentials](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/guides/azure_cli) and subsequent subscription to deploy the resources to.
 
-Currently only the `Basic` option for authenticating to the EST server using username and password is supported in these Terraform scripts.
\ No newline at end of file
+### Changing authentication mode
+The authentication mode is currently set at `Basic`, with default username for both the VM and EST set at `azureuser`, the passwords will be randomly generated and shown as outputs (together with the usernames) for testing purposes. If you would like to change these values you can provide other default values in the `variables.tf` files for both the `iot-edge` and `appservice` module. 
+
+If you want to use `x509` certificate-based authentication, then you would need to replace the default value of `auth_mode` within `/terraform/variables.tf` from "Basic" to "x509" and ensure that the `[cert_issuance.est.auth]` section in `cloud-init.yaml` looks like this:
+```
+      [cert_issuance.est.auth]
+      #username = "${EST_USERNAME}"
+      #password = "${EST_PASSWORD}"
+      
+      identity_cert = "file:///etc/aziot/estauth.pem"
+      identity_pk = "file:///etc/aziot/estauth.key.pem"
+```
+
+### Pushing code changes to Azure Container Registry
+If you make code changes locally and you are using Terraform to deploy your new code to the container image in Azure Container Registry, then you can ensure these changes are picked up by Terraform by removing the current state of the resource which builds the image using the following command: `terraform state rm 'module.acr.null_resource.push-docker'`
```

Comment:
What if we made `module.acr.null_resource.push-docker` to always execute (by adding a trigger in this resource with value [uuid()](https://www.terraform.io/language/functions/uuid). This way we won't have to worry about updating EST server code changes to the ACR.

Reply:
We were thinking the same indeed, but we figured it would be quite time consuming to rebuild the entire image every time you run `terraform apply`, but it would indeed remove the step of explicitly pushing the code changes to ACR. Which option has your preference?

Reply:
In an ideal world, the code change in EST would have its own release path outside of the infra deployment. In this case however, I think it's acceptable to rebuild the image every time.